e2e/cypress/integration/messaging/autocomplete_spec.js

```diff
@@ -0,0 +1,334 @@
+                cy.get('#suggestionList').within(() => {
+                    cy.findByText(displayNameTestUser).should('be.visible');
+                    cy.findByText(displayNameOtherUser).should('not.be.visible');
+                });
+            });
+
+            // # Post user mention
+            cy.get('#post_textbox').type('{enter}{enter}').then(() => {
+                // # Check that the user name has been posted
+                cy.getLastPostId().then((postId) => {
+                    cy.get(`#postMessageText_${postId}`).should('contain', `${testUser.username}`);
+                });
+            });
+        });
+    });
+
+    it('MM-T2203 @ autocomplete - not email', () => {
+        cy.visit(`/${testTeam.name}/channels/town-square`).wait(TIMEOUTS.FIVE_SEC).then(() => {
+            // # Clear then type @ and user email
+            cy.get('#post_textbox').should('be.visible').clear().type(`@${testUser.email}`).then(() => {
+                // * Verify that the item is displayed or not as expected.
+                cy.get('#suggestionList').should('not.be.visible');
+            });
+        });
+    });
+
+    it('MM-T2206 @ autocomplete - not in channel (center), have permission to add (public channel)', () => {
+        const message = `@${notInChannelUser.username} did not get notified by this mention because they are not in the channel. Would you like to add them to the channel? They will have access to all message history.`;
+
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`).wait(TIMEOUTS.FIVE_SEC).then(() => {
+            // # Clear then type @
+            cy.get('#post_textbox').should('be.visible').clear().type('@').then(() => {
+                // * Verify that the suggestion list is visible
+                cy.get('#suggestionList').should('be.visible');
+
+                // # Type user name
+                cy.get('#post_textbox').type(`${notInChannelUser.username}`);
+
+                // # Post user mention
+                cy.get('#post_textbox').type('{enter}{enter}').then(() => {
+                    cy.getLastPostId().then((postId) => {
+                        // * Verify that the correct system message is displayed or not
+                        cy.get(`#postMessageText_${postId}`).should('include.text', message);
+
+                        // * Click on the link to add the user to the channel
+                        cy.get('a.PostBody_addChannelMemberLink').should('be.visible').click().then(() => {
+                            // * Verify that the correct system message is displayed or not
+                            cy.getLastPostId().then((npostId) => {
+                                cy.get(`#postMessageText_${npostId}`).should('include.text', `@${notInChannelUser.username} added to the channel by you`);
+                            });
+                        });
+                    });
+                });
+            });
+        });
+    });
```

Comment:
@DHaussermann Thanks for running the tests.
@catalintomai I ran it again and got only 1 out of several runs with all tests passed; the rests randomly failed like what Dylan have seen. My recommendations would be:
1. To use `cy.uiWaitUntilMessagePostedIncludes` before doing `cy.getLastPostId` since other ephemeral post is not being posted before `cy.getLastPostId` gets executed.
2. To avoid `.then` as much as possible especially for commands that's not expecting return value/s, since the asynchronous nature of commands are being handled by [Cypress itself](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress.html#Commands-Are-Asynchronous). (nit, non-blocking, make more Cypress-y ;) and less Javascript pattern ) 

Something like,

```javascript
    it('MM-T2206 @ autocomplete - not in channel (center), have permission to add (public channel)', () => {
        const message = `@${notInChannelUser.username} did not get notified by this mention because they are not in the channel. Would you like to add them to the channel? They will have access to all message history.`;

        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`).wait(TIMEOUTS.FIVE_SEC);

        // # Clear then type @
        cy.get('#post_textbox').should('be.visible').clear().type('@');

        // * Verify that the suggestion list is visible
        cy.get('#suggestionList').should('be.visible');

        // # Type user name
        cy.get('#post_textbox').type(`${notInChannelUser.username}`);

        // # Post user mention
        cy.get('#post_textbox').type('{enter}{enter}');
        cy.uiWaitUntilMessagePostedIncludes('They will have access to all message history.');

        cy.getLastPostId().then((postId) => {
            // * Verify that the correct system message is displayed or not
            cy.get(`#postMessageText_${postId}`).should('include.text', message);

            // * Click on the link to add the user to the channel
            cy.get('a.PostBody_addChannelMemberLink').should('be.visible').click();
            cy.uiWaitUntilMessagePostedIncludes('added to the channel by you');

            // * Verify that the correct system message is displayed or not
            cy.getLastPostId().then((npostId) => {
                cy.get(`#postMessageText_${npostId}`).should('include.text', `@${notInChannelUser.username} added to the channel by you`);
            });
        });
    });
```

Reply:
Thanks @saturninoabril , I updated the code following your suggestions. The tests pass in my env (thought they were passing before as well), hopefully they'll pass on your envs. as well = )
cc: @DHaussermann